### PR TITLE
core: support old version of Page.getInstallabilityErrors

### PR DESCRIPTION
**Pull request by connorjclark**
Fixes https://github.com/GoogleChrome/lighthouse/issues/10318

### CODE REVIEW

File: lighthouse-core/computed/manifest-values.js [hunk 1]
```diff
@@ -43,14 +43,12 @@ class ManifestValues {
...
-            'Could not download a required icon from the manifest',
-            // kNoIconAvailableMessage
-            'Downloaded icon was empty or corrupted',
+          const failedToFetchIconErrorIds = [
+            'cannot-download-icon',
+            'no-icon-available',
           ];
           return icons.doExist(manifestValue) &&
-            !failedToFetchIconErrors.some(error => errors.includes(error));
+            !errors.some(error => failedToFetchIconErrorIds.includes(error.errorId));
```
Comment (patrickhulce) [inline lighthouse-core/computed/manifest-values.js]: sweet, this works out even more nicely for us :)

File: lighthouse-core/gather/gather-runner.js [hunk 2]
```diff
@@ -494,6 +494,39 @@ class GatherRunner {
...
+   */
+  static async getInstallabilityErrors(passContext) {
+    const response =
+      await passContext.driver.sendCommand('Page.getInstallabilityErrors');
+
+    let errors = response.installabilityErrors;
+    // Before M82, `getInstallabilityErrors` was not localized and just english
+    // error strings were returned. Convert the values we care about to the new error id format.
+    if (!errors) {
+      // @ts-ignore
```
Comment (patrickhulce) [inline lighthouse-core/gather/gather-runner.js]: how about
```suggestion
      /** @type {Array<string>} */
      // @ts-ignore - intentionally support older incompatible protocol data
      const m81StyleErrors = response.errors || [];
      errors = m81StyleErrors.map(error => {
```

primarily care about narrowing the scopre of the tsignore

File: types/artifacts.d.ts [hunk 3]
```diff
@@ -363,10 +363,14 @@ declare global {
           parentNode?: DomNodeMaybeWithParent;
         }
       }
-
+
```
Comment (patrickhulce) [inline types/artifacts.d.ts]: revert?

File: lighthouse-core/gather/gather-runner.js [hunk 4]
```diff
@@ -494,6 +494,39 @@ class GatherRunner {
     };
   }
 
+  /**
+   * Populates the important base artifacts from a fully loaded test page.
+   * Currently must be run before `start-url` gatherer so that `WebAppManifest`
+   * will be available to it.
+   * @param {LH.Gatherer.PassContext} passContext
+   * @return {LH.Artifacts.} passContext
```
Comment (patrickhulce) [inline lighthouse-core/gather/gather-runner.js]: hm that seems not right :)

File: lighthouse-core/gather/gather-runner.js [hunk 5]
```diff
@@ -494,6 +494,39 @@ class GatherRunner {
...
+   */
+  static async getInstallabilityErrors(passContext) {
```
Comment (patrickhulce) [inline lighthouse-core/gather/gather-runner.js]: gather runner tests for this method would be nice :D

File: lighthouse-core/gather/gather-runner.js (hunk 2 above)
Comment (connorjclark) [reply lighthouse-core/gather/gather-runner.js]: nice

File: lighthouse-core/gather/gather-runner.js (hunk 5 above)
Comment (connorjclark) [reply lighthouse-core/gather/gather-runner.js]: done

File: lighthouse-core/gather/gather-runner.js [hunk 6]
```diff
@@ -494,6 +494,41 @@ class GatherRunner {
     };
   }
 
+  /**
+   * Populates the important base artifacts from a fully loaded test page.
```
Comment (patrickhulce) [inline lighthouse-core/gather/gather-runner.js]: I think this jsdoc description is off ;)

Comment (connorjclark) [reply lighthouse-core/gather/gather-runner.js]: hehe

File: lighthouse-core/computed/manifest-values.js (hunk 1 above)
Comment (connorjclark) [reply lighthouse-core/computed/manifest-values.js]: yea i agree

**Pull request merged**